consensus-engine/src/utils.rs

```diff
@@ -0,0 +1,96 @@
+#[cfg(feature = "serde")]
+pub(crate) mod serde {
+    fn serialize_human_readable_bytes_array<const N: usize, S: serde::Serializer>(
+        src: [u8; N],
+        serializer: S,
+    ) -> Result<S::Ok, S::Error> {
+        use serde::Serialize;
+        use std::fmt::Write;
+        let size = N * 2 + 2;
+        let mut s = String::with_capacity(size);
+        s.push_str("0x");
+        for v in src {
+            // unwrap is safe because we allocate enough space
+            write!(&mut s, "{:02x}", v).unwrap();
+        }
+        s.serialize(serializer)
+    }
+
+    pub(crate) fn serialize_bytes_array<const N: usize, S: serde::Serializer>(
+        src: [u8; N],
+        serializer: S,
+    ) -> Result<S::Ok, S::Error> {
+        if serializer.is_human_readable() {
+            serialize_human_readable_bytes_array(src, serializer)
+        } else {
+            serializer.serialize_bytes(&src)
+        }
+    }
+
+    fn deserialize_human_readable_bytes_array<'de, const N: usize, D: serde::Deserializer<'de>>(
```

Comment:
there has to be a crate that does this (e.g. `hex`)

Reply:
Yeah, find a good one, switch to `const-hex`.